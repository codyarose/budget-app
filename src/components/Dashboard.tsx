import React, { FC } from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilter from '../features/filter/ExpenseListFilter'

const Dashboard: FC = () => {
	return (
		<div>
			<ExpenseListFilter />
			<ExpenseList />
		</div>
	)
}

export default Dashboard
