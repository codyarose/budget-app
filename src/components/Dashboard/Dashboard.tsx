import React, { FC, useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'

import ExpenseList from '../../features/expenses/ExpenseList'
import ExpenseListFilter from '../../features/filter/ExpenseListFilter'
import ExpenseTotal from '../../features/expenses/ExpensesTotal'
import {
	selectVisibleExpenses,
	selectTotal,
	setExpenses,
} from '../../features/expenses/expensesSlice'

const Dashboard: FC = () => {
	const dispatch = useDispatch()

	const expenses = useSelector(selectVisibleExpenses)
	const total = useSelector(selectTotal)

	useEffect(() => {
		dispatch(setExpenses())
	}, [dispatch])

	return (
		<div>
			<ExpenseTotal count={expenses.length} total={total} />
			<ExpenseListFilter />
			<ExpenseList expenses={expenses} />
		</div>
	)
}

export default Dashboard
