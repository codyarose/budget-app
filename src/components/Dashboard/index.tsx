import React, { FC } from 'react'
import ExpenseList from '../../features/expenses/ExpenseList'
import ExpenseListFilter from '../../features/filter/ExpenseListFilter'

const Dashboard: FC = () => {
	return (
		<div>
			<ExpenseListFilter />
			<ExpenseList />
		</div>
	)
}

export default Dashboard
