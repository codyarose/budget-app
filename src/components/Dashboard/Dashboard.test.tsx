import React from 'react'
import { shallow } from 'enzyme'
import Dashboard from './index'

describe('Dashboard component', () => {
	it('should render correctly', () => {
		const wrapper = shallow(<Dashboard />)

		expect(wrapper).toMatchSnapshot()
	})
})
