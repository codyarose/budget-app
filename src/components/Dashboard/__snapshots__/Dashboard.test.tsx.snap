// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component should render correctly 1`] = `
<div>
  <ExpenseListFilter />
  <ExpenseList />
</div>
`;
